File: rust/src/2022/day01.rs
// Part 1 + 2
use std::fs;

pub fn run() {
    let p_input: String = fs::read_to_string("data/2022/day01.txt").unwrap();
    let mut calories_by_elf: Vec<u32> = p_input
        .split("\n\n")
        .map(|block| block.lines().map(|line| line.parse::<u32>().unwrap()).sum())
        .collect();

    calories_by_elf.sort_unstable_by(|a, b| b.cmp(a));

    println!("{}", calories_by_elf[0]);
    println!("{}", calories_by_elf.iter().take(3).sum::<u32>());
}

File: python/2022/day01.py
# Python
## Part 1
with open("day01.txt") as f:
    input_calories = f.read().split("\n\n")

calories_by_elf = sorted(
    map(
        sum,
        [map(int, elf.splitlines()) for elf in input_calories],
    )
)

print(calories_by_elf[-1])

## Part 2
print(sum(calories_by_elf[-3:]))
